torch/csrc/jit/script/init.cpp

```diff
@@ -196,6 +196,35 @@ void checkOverloadDecl(const Decl& new_decl, const Decl& old_decl) {
 }
 } // namespace
 
+bool checkMutableFunctionDefault(const py::object& def_arg) {
+  if (py::isinstance<py::list>(def_arg) || py::isinstance<py::dict>(def_arg)) {
```

Comment:
We should also check for instances of classes

Reply:
that doesn't work right now anyway